packages/eslint-config-expo/flat/utils/typescript.js

```diff
@@ -80,6 +80,20 @@ module.exports = [
       'no-useless-constructor': 'off',
       '@typescript-eslint/no-useless-constructor': 'warn',
       'no-undef': 'off',
+
+      // Prevent use of CJS `require` syntax unless importing assets to align with Metro behavior.
+      '@typescript-eslint/no-require-imports': [
+        'warn',
+        {
+          // Allow supported asset extensions:
+          // https://github.com/facebook/metro/blob/9e1a6da5a7cd71bb9243f45644efe655870e5fff/packages/metro-config/src/defaults/defaults.js#L18-L53
+          // https://github.com/expo/expo/blob/c774cfaa7898098411fc7e09dcb409b7cb5064f9/packages/%40expo/metro-config/src/ExpoMetroConfig.ts#L247-L254
+          // Includes json which can be imported both as source and asset.
+          allow: [
+            '\\.(aac|aiff|avif|bmp|caf|db|gif|heic|html|jpeg|jpg|json|m4a|m4v|mov|mp3|mp4|mpeg|mpg|otf|pdf|png|psd|svg|ttf|wav|webm|webp|xml|yaml|yml|zip)$',
```

Comment:
How about we flip it the other way? Do not allow `.js`, `.jsx`, `.ts`, `.tsx`, `.json`, and `<no-extension>`. People might add more asset extensions through Metro - and this doesn't include that.

Reply:
Tried this but it's a bit harder to know if the module has no extension, especially given platform extensions.